Mathlib/NumberTheory/Cyclotomic/PID.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2024 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+
+import Mathlib.NumberTheory.NumberField.ClassNumber
+import Mathlib.NumberTheory.Cyclotomic.Rat
+import Mathlib.NumberTheory.Cyclotomic.Embeddings
+
+/-!
+# Cyclotomic fields that are PID.
+We prove that `ℤ [ζₚ]` is a PID for specific values of `p`. The result holds for `p ≤ 19`,
+but the proof is more and more involved.
+
+## Main results
+* `three_pid`: If `IsCyclotomicExtension {3} ℚ K` then `𝓞 K` is a principal ideal domain.
+* `five_pid`: If `IsCyclotomicExtension {5} ℚ K` then `𝓞 K` is a principal ideal domain.
+-/
+
+universe u
+
+namespace IsCyclotomicExtension.Rat
+
+open NumberField Polynomial InfinitePlace Nat Real cyclotomic
+
+variable (K : Type u) [Field K] [NumberField K]
+
+/-- If `IsCyclotomicExtension {3} ℚ K` then `𝓞 K` is a principal ideal domain. -/
+theorem three_pid [IsCyclotomicExtension {3} ℚ K] : IsPrincipalIdealRing (𝓞 K) := by
+  have hp : Fact (Nat.Prime ((3 : ℕ+) : ℕ)) := ⟨Nat.prime_three⟩
+  apply RingOfIntegers.isPrincipalIdealRing_of_abs_discr_lt
+  rw [absdiscr_prime 3 K, IsCyclotomicExtension.finrank (n := 3) K
+    (irreducible_rat (by norm_num)), nrComplexPlaces_eq_totient_div_two 3, totient_prime hp.1]
+  simp only [Int.reduceNeg, show ((3 : ℕ+) : ℕ) = 3 by rfl, succ_sub_succ_eq_sub, tsub_zero,
```

Comment:
There should be a `simp` lemma that does `((3 : ℕ+) : ℕ) = 3` for all positive values of `3`. But I cannot figure out how to spell it so that it works.

Reply:
Can you add
```lean
@[simp]
lemma PNat.val_ofNat (n : ℕ) : ((no_index (OfNat.ofNat n.succ) : ℕ+) : ℕ) = n.succ := rfl
```
at a suitable place and then use it here?

Reply:
Done, thanks for the suggestion!